src/exampleTypes/CheckableResultTable.js

```diff
@@ -2,26 +2,50 @@ import React from 'react'
 import _ from 'lodash/fp'
 import { observer } from 'mobx-react'
 import F from 'futil-js'
+import InjectTreeNode from '../utils/injectTreeNode'
+import { getResults } from '../utils/schema'
 
 // Extends ResultTable with a checkbox column
 // Writes to a lens called `selected`, using getValue to map the selected record to a value.
 // getValues uses _.iteratee, so it defaults to identity and supports things like strings to get props
-let CheckableResultTable = observer(
-  ({ fields, selected, getValue, Checkbox, ResultTable, ...props }) => (
-    <ResultTable
-      fields={{
-        _checkbox: {
-          label: <Checkbox />,
-          display: (x, y) => (
-            <Checkbox
-              {...F.domLens.checkboxValues(_.iteratee(getValue)(y), selected)}
-            />
-          ),
-        },
-        ...fields,
-      }}
-      {...props}
-    />
+let CheckableResultTable = InjectTreeNode(
+  observer(
+    ({ node, fields, selected, getValue, Checkbox, ResultTable, ...props }) => {
+      let results = getResults(node).slice()
+      let allChecked = results.length === F.view(selected).length
+      let checkAll = F.sets(
+        allChecked
+          ? []
+          : _.map(
+              _.flow(
+                _.get('_source'),
```

Comment:
We should use the getRecord thing since it won't always have _source (contexture-mongo doesn't)

Reply:
thanks! I forgot about that